tests/workflows/features/test_atlasbased.py

```diff
@@ -0,0 +1,161 @@
+    Checks that the matrixes are symmetric and positive-semidefinite
+    Checks coverage values are between 0 and 1
+    """
+
+    wf = init_atlas_based_connectivity_wf(
+        workdir=str(wd),
+        atlas_files=brainnetome_atlas,
+        atlas_spaces=["MNI152NLin2009cAsym"],
+    )
+
+    wf.base_dir = str(wd)
+    wf.inputs.inputnode.bold = func_file[0]
+    wf.inputs.inputnode.mask = func_file[1]
+    wf.inputs.resample.reference_image = brainnetome_atlas
+
+    assert wf.name == "atlas_based_connectivity_wf"
+    assert all(
+        node_name in [node.name for node in wf._graph.nodes()]
+        for node_name in ["inputnode", "make_resultdicts", "calcmean"]
+    ), "One or more expected nodes are missing"
+
+    logger.info(f"Brainnetome shape: {nib.nifti1.load(str(brainnetome_atlas)).shape}")
+    logger.info(f"Func file shape: {nib.nifti1.load(str(func_file[0])).shape}")
+    logger.info(f"Mask file shape: {nib.nifti1.load(str(func_file[1])).shape}")
+
+    run_workflow(wf)
+
+    # Corr matrix checks
+    corr_mat_path = wd / "grouplevel" / "func" / "desc-correlation_matrix.tsv"
+    atlas_img = nib.nifti1.load(str(brainnetome_atlas))
+    num_regions = (
+        len(np.unique(atlas_img.get_fdata())) - 1
+    )  # Subtract 1 to exclude background
+    correlation_matrix = np.loadtxt(corr_mat_path, delimiter="\t")
+    assert correlation_matrix.shape == (
+        num_regions,
+        num_regions,
+    ), "Correlation matrix shape does not match number of atlas regions"
+    assert np.allclose(
+        correlation_matrix, correlation_matrix.T, equal_nan=True
+    ), "Correlation matrix is not symmetric"
+    assert np.allclose(np.diag(correlation_matrix), 1), "Diagonal is not 1"
+    assert np.all(
+        (-1 <= correlation_matrix) & (correlation_matrix <= 1)
+    ), "Correlation matrix values are not between -1 and 1"
+
+    eigenvalues = np.linalg.eigvals(correlation_matrix)
+    assert np.all(
+        (eigenvalues > 0) | np.isclose(eigenvalues, 0)
+    ), "Correlation matrix is not positive-semidefinite"
+
+    # Cov matrix checks
+    cov_mat_path = wd / "grouplevel" / "func" / "desc-covariance_matrix.tsv"
+    covariance_matrix = np.loadtxt(cov_mat_path, delimiter="\t")
+    assert covariance_matrix.shape == (
+        num_regions,
+        num_regions,
+    ), "Covariance matrix shape does not match number of atlas regions"
+    assert np.allclose(
+        covariance_matrix, covariance_matrix.T
+    ), "Covariance matrix is not symmetric"
+
+    # Time series check
+    ts_path = wd / "grouplevel" / "func" / "timeseries.tsv"
+    ts = np.loadtxt(ts_path, delimiter="\t")
+    func_shape = nib.nifti1.load(str(func_file[0])).shape
+    assert (
+        func_shape[3] == ts.shape[0]
+    ), "Number of rows in timeseries.tsv does not match number of volumes"
+    assert (
+        num_regions == ts.shape[1]
+    ), "Number of columns in timeseries.tsv does not match number of atlas regions"
```

Comment:
Could you make sure that the correlation/covariance matrices actually correspond to the time series, for example using the `np.corrcoef` function?